### PR TITLE
qt: fix PayToEdit.parse_amount

### PR DESCRIPTION
**Pull request by zebra-lucky**
- qt: fix PayToEdit.parse_amount

### CODE REVIEW

**Comment by SomberNight** [comment]
You add more context when opening a PR.
I imagine the intention here is to change the error messages shown in the GUI?

-----

Scenario 1:
```
tb1qc5ztxm2kvtrtxun50v0rn6asm9tv0t3mfzh68v,asd
tb1qafmd8yfrvun27lt6nsg2hesqz2g4f666nx0dx4,0.002
```

Old GUI error dialog:
```
Invalid Lines found:

Line #1: tb1qc5ztxm2kvtrtxun50v0rn6asm9tv0t3mfzh6... (InvalidOperation([<class 'decimal.ConversionSyntax'>]))
```

New GUI error dialog:
```
Invalid Lines found:

Line #1: tb1qc5ztxm2kvtrtxun50v0rn6asm9tv0t3mfzh6... (Exception('Invalid amount'))
```

-----

Scenario 2:
```
tb1qc5ztxm2kvtrtxun50v0rn6asm9tv0t3mfzh68v,  
tb1qafmd8yfrvun27lt6nsg2hesqz2g4f666nx0dx4,0.002
```

Old GUI error dialog:
```
Invalid Lines found:

Line #1: tb1qc5ztxm2kvtrtxun50v0rn6asm9tv0t3mfzh6... (InvalidOperation([<class 'decimal.ConversionSyntax'>]))
```

New GUI error dialog:
```
Invalid Lines found:

Line #1: tb1qc5ztxm2kvtrtxun50v0rn6asm9tv0t3mfzh6... (Exception('Amount is empty'))
```

File: electrum/gui/qt/paytoedit.py [hunk 1]
```diff
@@ -127,10 +128,16 @@ def parse_script(self, x):
         return script
 
     def parse_amount(self, x):
-        if x.strip() == '!':
+        x_stripped = x.strip()
```
Comment (SomberNight) [inline electrum/gui/qt/paytoedit.py:131]: let's just do `x = x.strip()`

**Comment by zebra-lucky** [comment]
> You add more context when opening a PR.
> I imagine the intention here is to change the error messages shown in the GUI?

Yes.

File: electrum/gui/qt/paytoedit.py (hunk 1 above)
Comment (zebra-lucky) [reply electrum/gui/qt/paytoedit.py:131]: ok, I'll add changes in a half or hour

Comment (zebra-lucky) [reply electrum/gui/qt/paytoedit.py]: `x_stripped` -> `x`, rebased fix commit with squash

**Pull request merged**